bindings/matrix-sdk-crypto-ffi/src/dehydrated_devices.rs

```diff
@@ -73,20 +78,18 @@ impl DehydratedDevices {
         let device_data: Raw<_> = serde_json::from_str(&device_data)?;
         let device_id: OwnedDeviceId = device_id.into();
 
-        let mut key = get_pickle_key(&pickle_key)?;
+        let key: DehydratedDeviceKey = pickle_key.try_into()?;
 
         let ret = RehydratedDevice {
             runtime: self.runtime.to_owned(),
             inner: ManuallyDrop::new(self.runtime.block_on(self.inner.rehydrate(
-                &key,
+                key,
                 &device_id,
                 device_data,
             ))?),
         }
         .into();
 
-        key.zeroize();
```

Comment:
The `DehydratedDeviceKey` derives `ZeroizeOnDrop` so I believe this is not needed anymore.
I have a question regarding the `pickle_key` parameter, should not it be zeroized too?

Reply:
Both observations are correct.